docs/api/app.md

```diff
@@ -1,6 +1,16 @@
 # `app()`
 
-A function that initializes and mounts a Hyperapp application.
+**_Definition:_**
+
+> A function that initializes and mounts a Hyperapp application.
+
+**_Signature:_**
+
+```elm
+app : ({ Init, View, Node, Subscriptions?, Dispatch? }) -> DispatchFn
```

Comment:
Do these match the type definitions? We should be careful not to duplicate work.

Reply:
Not (yet). At the time of writing I clearly wasn't aware of the final (?) names for types. Besides that, the intention of these type annotations/definitions is to quickly give the reader an overview of what goes in and what comes out, without paying too much intention to the _exact math and inner workings and dependencies_. 

So, if you insist on having the _exact same_ names as in the type definitions (https://github.com/jorgebucaran/hyperapp/blob/main/index.d.ts) then I suggest to wait, till _those_ are finalized. Then we can retrofit them into the reference. But again: please not _too_ specific like in the index.d.ts, otherwise the reader will tend to ignore the annotation or be more confused than informed.

Or, we just keep them _light_ and just try to match them to the names used in the text.

In either case, we should be consistent. And, I would like to keep them (in one form or the other). I can be as insistive as you with :100: ;-).

Reply:
I don't know if they should be exactly the same, maybe a simplified, less accurate, subset of them would do, but we definitely don't want to create _new_ definitions.

Reply:
I don't know if they should be exactly the same, maybe a simplified, less accurate, subset of them would do, but we don't want to create _new_ definitions.

Basically, this means, that we are allowed to annotate only what has already been annotated using the names and symbols in the type definitions, but cannot add something new.

Reply:
I don't know if they should be exactly the same, maybe a simplified, less accurate, subset of them would do, but we don't want to create _new_ definitions.

Basically, this means that we are allowed to include only what has already been annotated, using the names and symbols in the type definitions, but not new stuff.

Reply:
Paraphrasing the type definitions where convenient is probably the way to go. While I did try keeping the type definitions readable as much as possible, they are first and foremost geared towards precision.